cmd/query/app/server.go

```diff
@@ -81,29 +84,27 @@ func NewServer(logger *zap.Logger, healthCheck *healthcheck.HealthCheck, querySv
 		return nil, errors.New("server with TLS enabled can not use same host ports for gRPC and HTTP.  Use dedicated HTTP and gRPC host ports instead")
 	}
 
-	grpcServer, err := createGRPCServer(querySvc, metricsQuerySvc, options, tm, logger, tracer)
+	grpcServer, err := createGRPCServer(querySvc, metricsQuerySvc, options, tm, telset.Logger, telset.TracerProvider)
 	if err != nil {
 		return nil, err
 	}
 
-	httpServer, err := createHTTPServer(querySvc, metricsQuerySvc, options, tm, tracer, logger)
+	httpServer, err := createHTTPServer(querySvc, metricsQuerySvc, options, tm, telset.TracerProvider, telset.Logger)
 	if err != nil {
 		return nil, err
 	}
 
 	return &Server{
-		logger:        logger,
-		healthCheck:   healthCheck,
 		querySvc:      querySvc,
 		queryOptions:  options,
-		tracer:        tracer,
 		grpcServer:    grpcServer,
 		httpServer:    httpServer,
 		separatePorts: grpcPort != httpPort,
+		telset:        telset,
 	}, nil
 }
 
-func createGRPCServer(querySvc *querysvc.QueryService, metricsQuerySvc querysvc.MetricsQueryService, options *QueryOptions, tm *tenancy.Manager, logger *zap.Logger, tracer *jtracer.JTracer) (*grpc.Server, error) {
+func createGRPCServer(querySvc *querysvc.QueryService, metricsQuerySvc querysvc.MetricsQueryService, options *QueryOptions, tm *tenancy.Manager, logger *zap.Logger, tracer trace.TracerProvider) (*grpc.Server, error) {
```

Comment:
accept telset here.

When calling NewGRPCHandler you don't actually need to pass a tracer at all, I checked the code and it's not being used. Which is sensible - the tracing is enabled at the grpc server level in this function, the handlers themselves don't need to do anything special. So let's clean that up - remove `GRPCHandler.tracer` field.